src/pylero/session.py

```diff
@@ -17,8 +18,35 @@
 
 
 logger = logging.getLogger(__name__)
+# We create a logger in order to intercept log records from suds.client
+suds_logger = logging.getLogger('suds.client')
+
 CERT_PATH = None
 
+# Regular expression to catch SOAP message containing password field
+REGEX_SOAP_MESSAGE_PASSWORD_FIELD = \
+        re.compile(r'(<ns1:password>)(.*)(</ns1:password>)')
+
+
+class ListenFilter(logging.Filter):
+    def filter(self, record):
+        """Determine which log records to output.
+        Returns 0 for no, nonzero for yes.
+        """
+        # We assume that this message contains the password in plaintext
+        # which as part of SOAP message
+        if '<ns1:password>' in record.getMessage():
+            masked_record = re.sub(REGEX_SOAP_MESSAGE_PASSWORD_FIELD,
+                                   r'\1*********\3',
+                                   record.getMessage())
+            logger.critical(masked_record)
+            return False
+        return True
+
+
+_filter = ListenFilter()
```

Comment:
- nit: any reason creating an instance rather than directly adding class?
- correct me if I'm wrong.

Reply:
@leelavg you are correct no need for creating a separate instance. 
Resolved this.